web/src/compose.js

```diff
@@ -779,14 +779,23 @@ export function reset_compose_scheduling_state(reset_edit_state = true) {
 }
 
 function schedule_message_to_custom_date() {
-    const request = create_message_object();
-    const selected_send_later_time = popover_menus.get_selected_send_later_time();
-    request.content = `/schedule ${selected_send_later_time}\n` + request.content;
+    const compose_message_object = create_message_object();
+    const scheduled_message_data = {
+        type: compose_message_object.type,
+        to: JSON.stringify(compose_message_object.to),
+        topic: compose_message_object.topic,
+        content: compose_message_object.content,
+        deliver_at: popover_menus.get_selected_send_later_time(),
+        tz_guess: new Intl.DateTimeFormat().resolvedOptions().timeZone,
+    };
+
     // If this is an edit request `scheduled_message_id` will be defined.
-    let scheduled_message_id;
     if ($("#compose-textarea").attr("data-scheduled-message-id")) {
-        scheduled_message_id = $("#compose-textarea").attr("data-scheduled-message-id");
+        scheduled_message_data.scheduled_message_id = $("#compose-textarea").attr(
+            "data-scheduled-message-id",
+        );
         $("#compose-textarea").removeAttr("data-scheduled-message-id");
```

Comment:
This line is probably in the wrong place and I think that's an important bug -- if the request fails, we'll have forgotten that we were planning to adjust an existing scheduled message and instead create a new one.

We need to do this in the `success` handler, probably.

Reply:
It looks like this is done in the `success` handler via: `compose.clear_compose_box()` -> `reset_compose_scheduling_state()` ->

```
export function reset_compose_scheduling_state(reset_edit_state = true) {
    $("#compose-textarea").prop("disabled", false);
    $("#compose-schedule-confirm-button").hide();
    popover_menus.reset_selected_schedule_time();
    $("#compose-send-button").show();
    if (reset_edit_state) {
        $("#compose-textarea").removeAttr("data-scheduled-message-id");
    }
}
```

So I think we can just delete this line.